src/training_hub/algorithms/lora_grpo_verl.py

```
@@ -0,0 +1,1077 @@
+
+logger = logging.getLogger(__name__)
+
+
+def _prepare_verl_data(
+    data_path: str,
+    output_dir: str,
+    n_train: int = 5000,
+    n_val: int = 500,
+    data_config: str = "Qwen3",
+    reward_type: str = "tool_call",
+    system_prompt: str = None,
```

Ruff (0.15.11)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)